clang/test/CIR/Transforms/complex-fold.cir

```diff
@@ -0,0 +1,44 @@
+// RUN: cir-opt --canonicalize -o %t.cir %s
```

Comment:
If you add `cir.complex.create` to `clang/lib/CIR/Dialect/Transforms/MergeCleanups.cpp`'s `runOnOperation`, you can also write this test in C/C++ (to make sure it works end-to-end)!

Reply:
Updated.